activerecord/lib/active_record/collection_cache_key.rb

```diff
@@ -7,17 +7,27 @@ def collection_cache_key(collection = all, timestamp_column = :updated_at) # :no
       if collection.loaded?
         size = collection.size
         if size > 0
-          timestamp = collection.max_by(&timestamp_column).public_send(timestamp_column)
+          timestamp = collection.max_by(&timestamp_column)._read_attribute(timestamp_column)
```

Comment:
This is not for performance but for consistency.
I feel this is a similar issue with #28660. If `collection` is not loaded, `timestamp` is an actual value from the database. But if `collection` is loaded, `timestamp` is affected by redefining the attribute getter.